public/app/features/alerting/unified/components/rule-editor/AlertRuleNameInput.tsx

```diff
@@ -63,16 +69,51 @@ export const AlertRuleNameAndMetric = () => {
           />
         </Field>
         {isGrafanaRecordingRule && (
-          <Field label="Metric" error={errors?.metric?.message} invalid={!!errors.metric?.message}>
-            <Input
-              id="metric"
-              width={38}
-              {...register('metric', {
-                required: { value: true, message: 'Must enter a metric name' },
-                pattern: recordingRuleNameValidationPattern(RuleFormType.grafanaRecording),
-              })}
-              aria-label="metric"
-              placeholder={`Give the name of the new recorded metric`}
+          <>
+            <Field label="Metric" error={errors?.metric?.message} invalid={!!errors.metric?.message}>
+              <Input
+                id="metric"
+                width={38}
+                {...register('metric', {
+                  required: { value: true, message: 'Must enter a metric name' },
+                  pattern: recordingRuleNameValidationPattern(RuleFormType.grafanaRecording),
+                })}
+                aria-label="metric"
+                placeholder={`Give the name of the new recorded metric`}
+              />
+            </Field>
+          </>
+        )}
+
+        {isGrafanaRecordingRule && config.featureToggles.grafanaManagedRecordingRulesDatasources && (
+          <Field
+            id="target-data-source"
+            label={t('alerting.recording-rules.label-target-data-source', 'Target data source')}
+            description={t(
+              'alerting.recording-rules.description-target-data-source',
+              'The Prometheus data source to store the recording rule in'
+            )}
+            error={errors.target_datasource_uid?.message}
+            invalid={!!errors.target_datasource_uid?.message}
+          >
+            <Controller
+              render={({ field: { onChange, ref, ...field } }) => (
+                <DataSourcePicker
+                  {...field}
+                  current={field.value}
+                  noDefault
+                  // Filter with `filter` prop instead of `type` prop to avoid showing the `-- Grafana --` data source
```

Comment:
I'm not sure if I understand this comment 🤔

Reply:
It's using the `filter` prop on the DataSourcePicker, as if we specify `type = prometheus` then we also see the built in Grafana datasource:

![image](https://github.com/user-attachments/assets/01a6a88e-cfee-490d-a3b1-09d2265cb77c)

Reply:
got it! thanks!